### PR TITLE
Feat/gateway login

### PR DESCRIPTION
**Pull request by isparling**
Blocked by merging `alpha:feat/cc-copy`; this is to consume endpoints exposed there.

### CODE REVIEW

File: src/Gateway.php [hunk 1]
```diff
@@ -15,6 +15,15 @@ public function __construct ( $host, $request_options = array())
         $this->description = 'Endpoint SDK for using the gateway';
     }
 
+    public function login($credentials) {
+        if (!is_array($credentials))
```
Comment (ahemphill) [inline src/Gateway.php]: I know the syntax is taken from existing functions in this package but I strongly prefer not excluding brackets

File: src/Gateway.php [hunk 2]
```diff
@@ -15,6 +15,15 @@ public function __construct ( $host, $request_options = array())
...
+            throw new Exception('Login requires first param to be an Associative Array');
```
Comment (ahemphill) [inline src/Gateway.php]: First but also only

File: src/Gateway.php [hunk 3]
```diff
@@ -15,6 +15,15 @@ public function __construct ( $host, $request_options = array())
...
+
+        $resp =  $this->post('v1/gateway/login', $credentials);
```
Comment (ahemphill) [inline src/Gateway.php]: Extra space

File: src/Gateway.php [hunk 4]
```diff
@@ -15,6 +15,15 @@ public function __construct ( $host, $request_options = array())
...
+        $jwt = $resp->headers['authorization'];
+        $this->setJwt($jwt);
```
Comment (ahemphill) [inline src/Gateway.php]: Is this not done after every response?

File: src/Gateway.php (hunk 1 above)
Comment (isparling) [reply src/Gateway.php]: Agree, will change for this PR.

File: src/Gateway.php (hunk 2 above)
Comment (isparling) [reply src/Gateway.php]: Copied from the same error in `Api->login()`; I'll update the exception there too.

File: tests/GatewayTest.php [hunk 5]
```diff
@@ -10,6 +10,11 @@
 class GatewayTest extends TestCase {
     const HOST = 'api.staging-rumbleship.com';
 
+    // required for testing overriding Api->login()
+    const TEST_ID_TOKEN= 'api123key';
```
Comment (ahemphill) [inline tests/GatewayTest.php]: Do these need to be at this level?

File: src/Gateway.php (hunk 4 above)
Comment (isparling) [reply src/Gateway.php]: Since we're overriding the inherited method, we have to re-implement the `setJwt()` call.

File: tests/GatewayTest.php [hunk 6]
```diff
@@ -200,4 +205,43 @@ function testGetConfig()
...
+    function testGatewayLogin()
+    {
+        $transport = new RequestToBodyMockTransport();
+        $gateway = new Gateway(self::HOST, array('transport' => $transport));
+        $credentials = array(
+            'id_token' => 'mylongidtokenasdfasdfasdf',
+            'email' => 'test@rumbleship.com'
+        );
+        $context = 'test-gateway';
+        $resp = $gateway->login($credentials, $context);
```
Comment (ahemphill) [inline tests/GatewayTest.php]: This does not align with the function

Comment (isparling) [reply tests/GatewayTest.php]: Oh good catch. Thanks!

File: tests/GatewayTest.php (hunk 5 above)
Comment (isparling) [reply tests/GatewayTest.php]: Nope. I don't expect other `GatewayTest` methods to require them; moved inline to test methods.

**Pull request merged**